feat.: Azure Docs (#20706)
newrelic/docs-website: src/content/docs/serverless-function-monitoring/azure-function-monitoring/env-variables-azure.mdx

  - Configuration
  - Secrets Management
---

> Environment variables are a way to store configuration and secrets outside of your code. The following tables display the mandatory variables and runtime specific variables that you can set to monitor your Lambda functions with New Relic.

Reviewer: ```suggestion
Environment variables are a way to store configuration and secrets outside of your code. The following tables display the mandatory variables and runtime specific variables that you can set to monitor your Azure functions with New Relic.
```

## Mandatory environment variables

> Regardless of your runtime, you must set the following environment variables to monitor your Lambda functions with New Relic:

Reviewer: ```suggestion
Regardless of your runtime, you must set the following environment variables to monitor your Azure functions with New Relic:
```

    </thead>
    <tbody>
        <tr>
            <td>`NODE_OPTIONS`</td>
            <td></td>
            <td>`-r newrelic`</td>
            <td>**Required**: Inject the agent before your Azure application functions are invoked. </td>
        </tr>
        <tr>
            <td>`NEW_RELIC_APP_NAME`</td>
            <td></td>
            <td></td>
            <td>Set the application name, though it is not used in the New Relic UI</td>
        </tr>
    </tbody>
</table>

You can find more environment variables in our [Node.js configuration documentation](/docs/apm/agents/nodejs-agent/installation-configuration/nodejs-agent-configuration/).
</TabsPageItem>

<TabsPageItem id="python">
<table>
    <thead>
        <tr>
            <th>Environment variable</th>
            <th>Default value</th>
            <th>Options</th>
            <th>Description</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>`NEW_RELIC_APP_NAME`</td>
            <td></td>
            <td></td>
            <td>Set the application name, though it is not used in the New Relic UI</td>
        </tr>
        <tr>
            <td>`FUNCTIONS_WORKER_RUNTIME`</td>
            <td></td>

>             <td>`“python”`</td>

Reviewer 2: ```suggestion
            <td>`"python"`</td>
```

            <td>Set the value to `python`.</td>
        </tr>
        <tr>
            <td>`PYTHON_ENABLE_WORKER_EXTENSIONS`</td>
            <td></td>
            <td>`1`</td>
            <td>Enable the python worker extension.</td>
        </tr>
        <tr>
            <td>`PYTHONPATH`</td>
            <td></td>
            <td>`${PYTHONPATH}:/home/site/wwwroot:/home/site/wwwroot/.python_packages/lib/site-packages`</td>

>             <td>Set the path for the your modules to directly imported into the Azure Function App</td>

Reviewer 2: ```suggestion
            <td>Add the installation target path of the modules installed from the `requirements.txt` file as a path to search when using import statements within the Azure Function App</td>
```

Based on your runtime, you can set the following environment variables to further configure the New Relic Lambda monitoring:

<Tabs>
  <TabsBar>
    <TabsBarItem id="nodejs">Node.js</TabsBarItem>
    <TabsBarItem id="python">Python</TabsBarItem>
    <TabsBarItem id="net">.NET</TabsBarItem>

  </TabsBar>

  <TabsPages>
    <TabsPageItem id="nodejs">
<table>
    <thead>
        <tr>
            <th>Environment variable</th>
            <th>Default value</th>
            <th>Options</th>
            <th>Description</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>`NODE_OPTIONS`</td>
            <td></td>
            <td>`-r newrelic`</td>
            <td>**Required**: Inject the agent before your Azure application functions are invoked. </td>
        </tr>
        <tr>
            <td>`NEW_RELIC_APP_NAME`</td>
            <td></td>
            <td></td>
            <td>Set the application name, though it is not used in the New Relic UI</td>
        </tr>
        <tr>
            <td>`NEW_RELIC_DISTRIBUTED_TRACING_ENABLED`</td>
            <td>`true`</td>
            <td>`true`, `false`</td>
            <td>Generate traces by enabling Distributed Tracing</td>

>         </tr>

Reviewer: Since the Node Agent's implementation of Distributed Tracing is on by default, this environment variable is required only if customers are turning DT off. I wouldn't expect it to be foregrounded in this configuration documentation.
